src/Configuration/ConfigurationRuleset.php

```diff
@@ -27,9 +29,35 @@ private function __construct(array $layerMap)
 
     /**
      * @return string[]
+     *
+     * @throws InvalidArgumentException
      */
     public function getAllowedDependencies(string $layerName): array
     {
-        return $this->layerMap[$layerName] ?? [];
+        return array_values(array_unique($this->getTransitiveDependencies($layerName, [])));
+    }
+
+    /**
+     * @param string[] $previousLayers
+     *
+     * @return string[]
+     *
+     * @throws InvalidArgumentException
+     */
+    private function getTransitiveDependencies(string $layerName, array $previousLayers): array
+    {
+        if (in_array($layerName, $previousLayers, true)) {
+            throw new InvalidArgumentException('Circular ruleset dependency for layer '.$layerName.' depending on: '.implode('->', $previousLayers));
+        }
+        $dependencies = [];
+        foreach ($this->layerMap[$layerName] ?? [] as $layer) {
+            if (0 === strncmp($layer, '+', 1)) {
```

Comment:
```suggestion
            if ('+' === $layer[0]) {
```
Should be faster, shouldn't it?

Reply:
Faster? Maybe. However, it is not binary-safe.

Reply:
Copied from PHP docs for Strings:
> Warning
Internally, PHP strings are byte arrays. As a result, accessing or modifying a string using array brackets is not multi-byte safe, and should only be done with strings that are in a single-byte encoding such as ISO-8859-1.

Reply:
I think there is no need to be binary safe for the first character. But feel free to use wether you are comfortable with

Reply:
@smoench my implementation is even used as a stub for `str_starts_with` before PHP8. That even has the advantage that once migrated to PHP8, manny tools will offer you an automatic upgrade. (like rectorphp/rector or PHPStorm.)